forms-flow-api/src/formsflow_api/services/form_process_mapper.py

```diff
@@ -103,19 +103,41 @@ def get_all_forms(  # pylint: disable=too-many-positional-arguments
             and CREATE_SUBMISSIONS in user.roles
             and ignore_designer
         ):
-            current_app.logger.debug("Fetching submissions count..")
-            for mapper in mappers_response:
-                mapper["submissionsCount"] = (
-                    Application.find_applications_count_by_parent_form_id_user(
-                        mapper["parentFormId"], user.user_name, user.tenant_key
-                    )
-                )
+            mappers_response = FormProcessMapperService.get_submissions_count(
+                mappers_response, user, sort_by, sort_order
+            )
 
         return (
             mappers_response,
             get_all_mappers_count,
         )
 
+    @classmethod
+    def get_submissions_count(
+        cls, mappers_response: List, user: UserContext, sort_by: List, sort_order: List
+    ):
+        """Get submissions count."""
+        current_app.logger.debug("Fetching submissions count..")
+        for mapper in mappers_response:
+            mapper["submissionsCount"] = (
```

Comment:
i think this will lead to N+1 problem, if we have 100 mappers data we can fetch the mapper data in a single query but fetching the application count it will take 100 query so instead of that we can query the data using subquery or join

Reply:
Refactored the code to utilize subqueries to fetch submission counts.